vsphere/host_virtual_switch_structure.go

```diff
@@ -0,0 +1,151 @@
+package vsphere
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// schemaHostVirtualSwitchBondBridge returns schema items for resources that
+// need to work with a HostVirtualSwitchBondBridge, such as virtual switches.
+func schemaHostVirtualSwitchBondBridge() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		// HostVirtualSwitchBeaconConfig
+		"beacon_interval": &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Determines how often, in seconds, a beacon should be sent to probe for the validity of a link.",
+			Default:     1,
+		},
+
+		// LinkDiscoveryProtocolConfig
+		"link_discovery_operation": &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Whether to advertise or listen for link discovery. Valid values are advertise, both, listen, and none.",
+			Default:     "listen",
+		},
+		"link_discovery_protocol": &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The discovery protocol type. Valid values are cdp and lldp.",
+			Default:     "cdp",
+		},
+
+		// HostVirtualSwitchBondBridge
+		"network_adapters": &schema.Schema{
+			Type:        schema.TypeList,
+			Required:    true,
+			Description: "The link discovery protocol configuration for the virtual switch.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+	}
+}
+
+// expandHostVirtualSwitchBeaconConfig reads certain ResourceData keys and
+// returns a HostVirtualSwitchBeaconConfig.
+func expandHostVirtualSwitchBeaconConfig(d *schema.ResourceData) *types.HostVirtualSwitchBeaconConfig {
+	obj := &types.HostVirtualSwitchBeaconConfig{
+		Interval: int32(d.Get("beacon_interval").(int)),
+	}
+	return obj
+}
+
+// flattenHostVirtualSwitchBeaconConfig reads various fields from a
+// HostVirtualSwitchBeaconConfig into the passed in ResourceData.
+func flattenHostVirtualSwitchBeaconConfig(d *schema.ResourceData, obj *types.HostVirtualSwitchBeaconConfig) error {
+	d.Set("beacon_interval", obj.Interval)
+	return nil
+}
+
+// expandLinkDiscoveryProtocolConfig reads certain ResourceData keys and
+// returns a LinkDiscoveryProtocolConfig.
+func expandLinkDiscoveryProtocolConfig(d *schema.ResourceData) *types.LinkDiscoveryProtocolConfig {
+	obj := &types.LinkDiscoveryProtocolConfig{
+		Operation: d.Get("link_discovery_operation").(string),
+		Protocol:  d.Get("link_discovery_protocol").(string),
```

Comment:
Hey @tombuildsstuff, as mentioned [above](https://github.com/terraform-providers/terraform-provider-vsphere/pull/138#discussion_r135106704) I'd like to stay away from nested values there's better first-class support for general nested config modeling in TF. This does not need a list or set either to support multiple values so I'm just leaving it flat.